### PR TITLE
MM-17888 Check plugin Helpers minimum server version comments

### PR DESCRIPTION
**Pull request by pbitty**
#### Summary

This implements a CI check that ensures `Minimum server version` comments in the `plugin.Helpers` methods match the effective minimum version of all the `API` calls that they make.

#### Ticket Link

Resolves https://github.com/mattermost/mattermost-server/issues/11912.

### CODE REVIEW

**Comment by pbitty** [comment]
Sorry, this is a big PR.  I've tried to break up each change into a logical commit, so that it's easier to review.  I got pretty deep into this and I think it's time to get an outside perspective.

**Comment by pbitty** [comment]
Sorry for the force-push.  I was trying to get the commits to appear in the right order after doing some rebase cleanup earlier.  I am done now.

File: plugin/checker/check_api.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package main
```
Comment (pbitty) [inline plugin/checker/check_api.go]: This is the logic that was formerly in `main.go`.

**Comment by pbitty** [comment]
Good question!  No, they are not transitive.  That's an interesting problem.  I am happy to try to implement that, but it might be worth doing it in another Issue and/or PR.

What do you think?

File: plugin/checker/check_helpers.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+			res.Errors = append(res.Errors, renderWithFilePosition(
+				fset,
+				pos,
+				fmt.Sprintf("documented minimum server version too low on method %s", name)),
+			)
+		} else {
+			res.Warnings = append(res.Warnings, renderWithFilePosition(
+				fset,
+				pos,
+				fmt.Sprintf("documented minimum server version too high on method %s", name)),
```
Comment (cpoile) [inline plugin/checker/check_helpers.go]: Is this something we want to enforce? (Just wondering if there may be a non-API reason why a function might be labelled higher than what is strictly required.)

**Comment by pbitty** [comment]
No problem!  I'm happy this is working.

@lieut-data , if you would like to make this a transitive check, I am happy to give it a shot.  If you decide to create an Issue for it, feel free to assign it to me.

File: plugin/checker/check_helpers.go (hunk 2 above)
Comment (lieut-data) [reply plugin/checker/check_helpers.go]: Yeah, my thoughts were that a helper might want to require a higher version in which a fix to an API was made, hence the warning. It's a purely theoretical concern right now.

**Comment by lieut-data** [comment]
@pbitty, I think this is great as-is -- let's let it soak for a while, and we can revisit the transitive idea separately.

**Comment by prapti** [comment]
@lieut-data does this PR need separate tests or does https://mattermost.atlassian.net/browse/MM-17889 cover this one as well?

**Comment by lieut-data** [comment]
@prapti, looks like this PR already has separate tests: https://github.com/mattermost/mattermost-server/pull/12663/files#diff-fa60ffe40f271fe40d1e0ede8605016b.

**Comment by prapti** [comment]
@lieut-data great!
Do you think additional QA verification needed? I'm looking at the [Jira ticket](https://mattermost.atlassian.net/browse/MM-17888) linked to this PR's [issue](https://github.com/mattermost/mattermost-server/issues/11912 and not sure if QA testing is needed, mostly because minimum server version comments was verified for https://mattermost.atlassian.net/browse/MM-17889.

**Comment by lieut-data** [comment]
@prapti, this is a subtly different improvement, in which we're adding these comments to the plugin helpers specifically, not just the API. I do think it's still useful for QA verification.

**Pull request merged**